### PR TITLE
[FIX] sale_automatic_workflow: Currency from Invoice

### PR DESCRIPTION
**Pull request by SodexisTeam**
When payment is created, the currency_id is not obtained from invoice instead the value is taken from company default.

### CODE REVIEW

**Comment by rousseldenis** [comment]
@SodexisTeam Thanks for this. Could you add a test for that change ?

**Comment by atchuthan** [comment]
> @SodexisTeam Thanks for this. Could you add a test for that change ?

Sure, we will add it to test file

**Comment by atchuthan** [comment]
> @SodexisTeam Thanks for this. Could you add a test for that change ?

@rousseldenis Testing added.  Ready for review.

**Comment by atchuthan** [comment]
> > @SodexisTeam Thanks for this. Could you add a test for that change ?
> 
> @rousseldenis Testing added. Ready for review.

@rousseldenis waiting for review, thanks

File: sale_automatic_workflow_payment_mode/models/automatic_workflow_job.py
Comment (rousseldenis) [inline sale_automatic_workflow_payment_mode/models/automatic_workflow_job.py:24]: @SodexisTeam Apart this is correct, note that if payment_mode_id is False, ORM will compute result of invoice.payment_mode_id.fixed_journal_id as False too.

So, this change is not necessary

Comment (atchuthan) [reply sale_automatic_workflow_payment_mode/models/automatic_workflow_job.py:24]: For the condition `not invoice.payment_mode_id.fixed_journal_id`, if there is no payment_mode_id, this condition returns True.  Thus, the payment will not be created for invoices that do not have payment_mode_id selected.

Comment (atchuthan) [reply sale_automatic_workflow_payment_mode/models/automatic_workflow_job.py:24]: @rousseldenis please review my last comment

**Comment by atchuthan** [comment]
@OCA/crm-sales-marketing-maintainers This PR is ready for review.  Thanks

**Review by rousseldenis — APPROVED** [review]
Code review

**Review by atchuthan — APPROVED** [review]
LGTM :+1:

**Review by amkarthik — APPROVED** [review]

**Pull request merged**
